sql/src/test/java/io/crate/operation/collect/sources/NodeStatsCollectSourceTest.java

```diff
@@ -115,12 +121,75 @@ public void testFilterNodes() throws Exception {
         Collection<CrateCollector> collectors = nodeStatsCollectSource.getCollectors(collectPhase, null, null);
         assertThat(collectors.size(), is(1));
 
-        // check that the collector has only the requested node
-        NodeStatsCollector collector = (NodeStatsCollector)collectors.iterator().next();
-        final Field collectorNodes = NodeStatsCollector.class.getDeclaredField("nodes");
-        collectorNodes.setAccessible(true);
-        List<DiscoveryNode> discoveryNodes = (List<DiscoveryNode>)collectorNodes.get(collector);
+        CrateCollector collector = collectors.iterator().next();
```

Comment:
would it make sense to test the node filtering more isolated - without involving the Collector?